### ISSUE TITLE
Support for subclasses of supported classifiers

### ISSUE THREAD

**Issue by zoj613** [shap/shap#863]
Shap's tree explainer does not seem to work at all for "custom" classifiers based to the supported ones.

Say I have:
```python
# modelling.py
from  sklearn.ensemble.gradient_boosting import GradientBoostingClassifier
class MyCustom(GradientBoostingClassifier, SomeMixin):
    def custom_func(self, ...):
        pass
   # other methods
```
If i used the shap `TreeExplainer` I get the error:
`AssertionError: <class 'src.modelling.MyCustom'> is not currently a supported model type!`

Is there a way to support custom classifiers that are subclasses of the supported ones if very minimal changes to the parent class are made??

Looking at the source code it seems like the string output of `str(type(model))` is the check that determines if an exception should be thrown:
https://github.com/slundberg/shap/blob/e475795cbd88fb5c0fc9b110540b872ce8456e4c/shap/explainers/tree.py#L407-L664

Maybe there could be something done to see if the custom classifier has inherited from the supported parent instead?

**Comment by zoj613** [comment]
ping @slundberg

**Comment by peterdhansen** [comment]
Is there any reason not to apply the fix in #863 to the plotting functions as well?
So if I have a pandas dataframe subclass (e.g. geopandas), it will still be recognized?
